uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequestCalendarFeedConfig.php

```diff
@@ -430,4 +431,62 @@ public function getInaccessibleFeedConfigs(
 
     return $inaccessibleFeeds;
   }
+
+  /**
+   * Returns a LeaveRequestCalendarFeedConfig object if an
+   * active feed config with the given hash is found.
+   *
+   * @param string $hash
+   *
+   * @return LeaveRequestCalendarFeedConfig
+   */
+  public static function findActiveByHash($hash) {
+    $feedConfig = new self();
+    $feedConfig->is_active = 1;
+    $feedConfig->hash = $hash;
+
+    if (!$feedConfig->find(TRUE)) {
+      throw new InvalidLeaveRequestCalendarFeedConfigException(
+        "An enabled feed with the given hash does not exist!"
+      );
+    }
+
+    return $feedConfig;
+  }
+
+  /**
+   * Returns the departments the feed config has set for the
+   * composed of property.
+   *
+   * @return string
+   */
+  public function getDepartmentsComposedOf() {
```

Comment:
Do we really need the `ComposedOf` as part of the method name? Why not simply `getDepartments()`, `getLocations()` etc? It doesn't matter for the other classes that this information is stored as an array inside `composed_of`

Reply:
There are departments and locations also stored in the visible_to property. So we need to distinguish between the departments the feed is composed of and the departments the feed link is visible to.